sdk/endpoint/ipam.go

```diff
@@ -51,12 +51,24 @@ func (ice *IpamEndpoint) Request(ctx context.Context, request *networkservice.Ne
 		return nil, err
 	}
 
+	/* Exclude the prefixes from the pool of available prefixes */
+	excludedPrefixes, err := ice.prefixPool.ExcludePrefixes(request.Connection.Context.ExcludedPrefixes)
+	if err != nil {
+		return nil, err
+	}
+
 	//TODO: We need to somehow support IPv6.
 	srcIP, dstIP, prefixes, err := ice.prefixPool.Extract(request.Connection.Id, connectioncontext.IpFamily_IPV4, request.Connection.Context.ExtraPrefixRequest...)
 	if err != nil {
 		return nil, err
 	}
 
+	/* Release the actual prefixes that were excluded during IPAM */
+	err = ice.prefixPool.ReleaseExcludedPrefixes(excludedPrefixes)
```

Comment:
And restore that copy right here? 
Then we can avoid implementing `ReleaseExcludePrefixes` method. What do you think about it?

Reply:
Make sense, but if we restore the original state of `prefixes` at this point - this happens right after we do the IPAM using the altered copy with the excluded prefixes. If successful, this results in extracting prefixes from it, so if we do that, in the end, we should have the troubles to take into account that change and translate it back to the original, unaltered copy.

Reply:
Yep, you're absolutely right!